acme/client.py

```diff
@@ -0,0 +1,234 @@
+"""Client object."""
+import requests
+import hashlib as hl
+import json
+import os
+
+VALID_METHODS = ['POST', 'GET', 'DELETE', 'PUT']
+
+class Client:
+    
+    def __init__(self, base_url, email=None, password=None):
+        """ME.org Client object.
+
+        Suppying email and password will automatically log in.
+
+        Parameters
+        ----------
+        base_url : str
+            Base URL to API.
+        email : str, optional
+            Registered email address, by default None
+        password : str, optional
+            User password, by default None
+        """
+        self.base_url = base_url
+        self.headers = dict()
+
+        # Automatically login.
+        if email is not None and password is not None:
+            self.login(email, password)
+
+
+    def _make_request(self, method: str, endpoint: str, data: dict=None, headers: dict=None) -> dict:
```

Comment:
We can use the [`typing.Union`](https://docs.python.org/3/library/typing.html#typing.Union) type here:
```suggestion
    def _make_request(self, method: str, endpoint: str, data: dict=None, headers: dict=None) -> Union[dict, requests.Response]:
```